src/events/etcd.py

```diff
@@ -769,5 +770,5 @@ def _update_client_relations(self) -> None:
                 self.charm.external_clients_manager.update_client_relations_data(
                     etcd_version=self.charm.cluster_manager.get_version()
                 )
-            except KeyError as e:
-                logger.warning(f"Error updating client relations data: {e}")
+            except (KeyError, PydanticSerializationError) as e:
```

Comment:
This is fine for the time being, but we should not have to catch this one here. 
This is the indication of a bigger problem, that we are getting unexpected payloads, we should figure out what is happening